modules/siddhi-extensions/event-table/src/main/java/org/wso2/siddhi/extension/eventtable/HazelcastEventTable.java

```diff
@@ -0,0 +1,308 @@
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.event.ComplexEvent;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.MetaComplexEvent;
+import org.wso2.siddhi.core.event.stream.MetaStreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.event.stream.StreamEventPool;
+import org.wso2.siddhi.core.event.stream.converter.ZeroStreamEventConverter;
+import org.wso2.siddhi.core.exception.OperationNotSupportedException;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.core.table.EventTable;
+import org.wso2.siddhi.core.util.SiddhiConstants;
+import org.wso2.siddhi.core.util.collection.operator.Finder;
+import org.wso2.siddhi.core.util.collection.operator.Operator;
+import org.wso2.siddhi.extension.eventtable.hazelcast.HazelcastEventTableConstants;
+import org.wso2.siddhi.extension.eventtable.hazelcast.HazelcastOperatorParser;
+import org.wso2.siddhi.extension.eventtable.hazelcast.internal.ds.HazelcastEventTableServiceValueHolder;
+import org.wso2.siddhi.query.api.annotation.Annotation;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.TableDefinition;
+import org.wso2.siddhi.query.api.expression.Expression;
+import org.wso2.siddhi.query.api.util.AnnotationHelper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Hazelcast event table implementation of SiddhiQL
+ */
+public class HazelcastEventTable implements EventTable {
+    private static final Logger logger = Logger.getLogger(HazelcastEventTable.class);
+    private final ZeroStreamEventConverter eventConverter = new ZeroStreamEventConverter();
+    private TableDefinition tableDefinition;
+    private ExecutionPlanContext executionPlanContext;
+    private StreamEventCloner streamEventCloner;
+    private StreamEventPool streamEventPool;
+    private HazelcastInstance hcInstance;
+    private ConcurrentMap<Object, StreamEvent> eventsMap = null;
+    private List<StreamEvent> eventsList = null;
+    private String indexAttribute = null;
+    private int indexPosition;
+    private String elementId;
+
+    /**
+     * Event Table initialization method, it checks the annotation and do necessary pre configuration tasks.
+     *
+     * @param tableDefinition      Definition of event table
+     * @param executionPlanContext ExecutionPlan related meta information
+     */
+    @Override
+    public void init(TableDefinition tableDefinition, ExecutionPlanContext executionPlanContext) {
+        this.tableDefinition = tableDefinition;
+        this.executionPlanContext = executionPlanContext;
+
+        Annotation fromAnnotation = AnnotationHelper.getAnnotation(SiddhiConstants.ANNOTATION_FROM, tableDefinition.getAnnotations());
```

Comment:
@dilini-muthumala in-order to use hazelcast event table, user has to define it as;
```
@from(eventtable = 'hazelcast')
define table OutputStream (symbol string, price int, volume float);
```
And if the `@from` annotation is not present, it'll function as a normal In-memory event table.